### PR TITLE
1965 deserialize credentials array

### PR DESCRIPTION
**Pull request by mssalvatore**
# What does this PR do?

Adds a convenience method to deserialize an array of credentials from JSON.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] ~Is the TravisCI build passing?~
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] ~Added relevant unit tests?~
* [ ] Have you successfully tested your changes locally? Elaborate:
    > Tested by running unit tests
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: monkey/common/credentials/credentials.py [hunk 1]
```diff
@@ -139,6 +162,41 @@ def from_json(credentials: str) -> Credentials:
         except MarshmallowError as err:
             raise InvalidCredentialsError(str(err))
 
+    @staticmethod
+    def from_json_array(credentials_array_json: str) -> Sequence[Credentials]:
```
Comment (VakarisZ) [inline monkey/common/credentials/credentials.py:166]: This seems redundant with the `from_json`. Whoever sends a single credential object can also put that into an array.

Comment (ilija-lazoroski) [reply monkey/common/credentials/credentials.py:166]: Yep but if you need to do it in a multiple places then it becomes code duplication.

Comment (mssalvatore) [reply monkey/common/credentials/credentials.py:166]: Yeah, it's for convenience and to reduce code duplication. If it turns out we don't use it, we can remove it.

Comment (VakarisZ) [reply monkey/common/credentials/credentials.py:166]: It's not a responsibility of the credentials dataclass to solve code duplication in the callers, it's to provide a good interface. I also don't understand how redundancy is better than duplication.

Comment (VakarisZ) [reply monkey/common/credentials/credentials.py:166]: The logic is out of place here, IMO. This is a credentials object definition, it's not about parsing json lists.

**Pull request merged**